src/models/UserMapper.php

```diff
@@ -385,4 +385,80 @@ protected function markUserVerified($user_id) {
         $verify_stmt->execute($verify_data);
         return true;
     }
+
+    /**
+     * Does the currently-authenticated user have right to edit this user?
+     *
+     * User must be the edited user, or a site admin
+     *
+     * @param int $user_id The identifier for the user to edit
+     * @return bool True if the user has privileges, false otherwise
+     */
+    public function thisUserHasAdminOn($user_id) {
+        // do we even have an authenticated user?
+        $loggedInUser = $this->_request->getUserId();
+        if(false !== $loggedInUser) {
```

Comment:
`$this->_request->getUserId()` will return null or a number. I don't think it will ever return a false?
You can remove the "if (false"  test anyway as the normal path through this method is a valid logged in user.